File: sdk-android-library/src/main/java/com/xyfindables/sdk/actionHelper/XYLight.java
package com.xyfindables.sdk.actionHelper;

import android.bluetooth.BluetoothGatt;
import android.bluetooth.BluetoothGattCharacteristic;
import android.util.Log;

import com.xyfindables.core.XYBase;
import com.xyfindables.sdk.XYDevice;
import com.xyfindables.sdk.action.XYDeviceActionGetLED;
import com.xyfindables.sdk.action.XYDeviceActionSetLED;

/**
 * Created by alex.mcelroy on 9/29/2017.
 */

public class XYLight extends XYActionHelper {

    private static final String TAG = XYLight.class.getSimpleName();

    public interface Callback extends XYActionHelper.Callback {
        void started(boolean success, byte[] value);
    }

    public XYLight(XYDevice device, final Callback callback) {
        action = new XYDeviceActionGetLED(device) {
            @Override
            public boolean statusChanged(int status, BluetoothGatt gatt, BluetoothGattCharacteristic characteristic, boolean success) {
                Log.v(TAG, "statusChanged:" + status + ":" + success);
                boolean result = super.statusChanged(status, gatt, characteristic, success);
                switch (status) {
                    case STATUS_CHARACTERISTIC_READ:
                        callback.started(success, value);
                        break;
                    case STATUS_COMPLETED:
                        callback.completed(success);
                        break;
                }
                return result;
            }
        };
    }

    public XYLight(XYDevice device, int value, final Callback callback) {
        action = new XYDeviceActionSetLED(device, value) {
            @Override
            public boolean statusChanged(int status, BluetoothGatt gatt, BluetoothGattCharacteristic characteristic, boolean success) {
                Log.v(TAG, "statusChanged:" + status + ":" + success);
                boolean result = super.statusChanged(status, gatt, characteristic, success);
                switch (status) {
                    case STATUS_COMPLETED:
                        callback.completed(success);
                        break;
                }
                return result;
            }
        };
    }
}

File: sdk-android-library/src/main/java/com/xyfindables/sdk/actionHelper/XYLight.kt
package com.xyfindables.sdk.actionHelper

import android.bluetooth.BluetoothGatt
import android.bluetooth.BluetoothGattCharacteristic
import com.xyfindables.sdk.XYDevice
import com.xyfindables.sdk.action.XYDeviceAction
import com.xyfindables.sdk.action.XYDeviceActionGetLED
import com.xyfindables.sdk.action.XYDeviceActionSetLED

/**
 * Created by alex.mcelroy on 9/29/2017.
 */

class XYLight : XYActionHelper {

    interface Callback : XYActionHelper.Callback {
        fun started(success: Boolean, value: ByteArray)
    }

    constructor(device: XYDevice, callback: Callback) {
        action = object : XYDeviceActionGetLED(device) {
            override fun statusChanged(status: Int, gatt: BluetoothGatt?, characteristic: BluetoothGattCharacteristic?, success: Boolean): Boolean {
                logExtreme(TAG, "statusChanged:$status:$success")
                val result = super.statusChanged(status, gatt, characteristic, success)
                when (status) {
                    XYDeviceAction.STATUS_CHARACTERISTIC_READ -> callback.started(success, value!!)
                    XYDeviceAction.STATUS_COMPLETED -> callback.completed(success)
                }
                return result
            }
        }
    }

    constructor(device: XYDevice, value: Int, callback: Callback) {
        action = object : XYDeviceActionSetLED(device, value) {
            override fun statusChanged(status: Int, gatt: BluetoothGatt?, characteristic: BluetoothGattCharacteristic?, success: Boolean): Boolean {
                logExtreme(TAG, "statusChanged:$status:$success")
                val result = super.statusChanged(status, gatt, characteristic, success)
                when (status) {
                    XYDeviceAction.STATUS_COMPLETED -> callback.completed(success)
                }
                return result
            }
        }
    }

    companion object {

        private val TAG = XYLight::class.java.simpleName
    }
}
